homeassistant/components/travel_time/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""The travel_time integration."""
+from datetime import timedelta
+import logging
+from typing import Dict, Optional, Union
+
+import voluptuous as vol
+
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import State
+from homeassistant.helpers import location
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.loader import bind_hass
+
+from .const import (
+    ATTR_DESTINATION,
+    ATTR_DESTINATION_NAME,
+    ATTR_DISTANCE,
+    ATTR_DURATION,
+    ATTR_DURATION_IN_TRAFFIC,
+    ATTR_MODE,
+    ATTR_ORIGIN,
+    ATTR_ORIGIN_NAME,
+    ATTR_ROUTE,
+    ATTR_ROUTE_MODE,
+    ATTR_TRAFFIC_MODE,
+    ATTR_UNIT_SYSTEM,
+    DOMAIN,
+    UNIT_OF_MEASUREMENT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Track states and offer events for sensors."""
+    component = hass.data[DOMAIN] = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL
+    )
+
+    await component.async_setup(config)
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a config entry."""
+    return await hass.data[DOMAIN].async_setup_entry(entry)
+
+
+async def async_unload_entry(hass, entry):
+    """Unload a config entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class TravelTimeEntity(Entity):
+    """Representation of a travel_time entity."""
+
+    @property
+    def attribution(self) -> str:
+        """Get the attribution of the travel_time entity."""
+        return None
+
+    @property
+    def destination(self) -> str:
+        """Get the destination coordinates of the travel_time entity."""
+        return None
+
+    @property
+    def destination_name(self) -> str:
+        """Get the destination name of the travel_time entity."""
+        return None
+
+    @property
+    def distance(self) -> str:
+        """Get the distance of the travel_time entity."""
+        return None
+
+    @property
+    def duration(self) -> str:
+        """Get the duration without traffic of the travel_time entity."""
+        return None
+
+    @property
+    def duration_in_traffic(self) -> str:
```

Comment:
Is this something that is always available ? Should we start with not having this

Reply:
Distance and destination_name are used by the travel_time card by @ljmerza.
Having both duration and duration_in_traffic enables certain usecases without having to create two sensors. E.g. alert me when the current duration_in_traffic is more than 30% of the normal duration.
waze_travel_time is the only integration to currently not support duration_in_traffic.

Reply:
ok let's keep it.